Extension/src/Debugger/debugAdapterDescriptorFactory.ts

```diff
@@ -9,32 +9,51 @@ import * as path from 'path';
 import * as os from 'os';
 
 // Registers DebugAdapterDescriptorFactory for `cppdbg` and `cppvsdbg`. If it is not ready, it will prompt a wait for the download dialog.
-// Note: util.extensionContext.extensionPath is needed for the commands because VsCode does not support relative paths for adapterExecutableComand
+// NOTE: This file is not automatically tested.
 
-export class CppdbgDebugAdapterDescriptorFactory implements vscode.DebugAdapterDescriptorFactory {
+abstract class DebugAdapterDescriptorFactoryWithContext implements vscode.DebugAdapterDescriptorFactory {
+    protected readonly context: vscode.ExtensionContext;
+
+    // This is important for the Mock Debugger since it can not use src/common
+    constructor(context: vscode.ExtensionContext) {
+        this.context = context;
```

Comment:
Could this be stale at any point in that it might change after activation but since we are caching it, we might have a stale one?

Reply:
It is the same one that VS Code passes in when we activate, if its stale then that's how VSCode handles it.